Sources/StreamChat/Utils/Database/NSManagedObject+Extensions.swift

```diff
@@ -10,13 +10,109 @@ extension NSManagedObject {
     }
 }
 
+extension NSEntityDescription {
+    class func insertNewObject<T: NSManagedObject>(
+        into context: NSManagedObjectContext,
+        for request: NSFetchRequest<T>
+    ) -> T {
+        let obj = T(context: context)
+        request.entity = obj.entity
+        FetchCache.shared.set(request, objectIds: [obj.objectID])
+        return obj
+    }
+}
+
+protocol NSFetchRequestGettable {}
+
+private let idKey = "id"
+
+extension NSFetchRequestGettable where Self: NSManagedObject {
+    static func fetchRequest(id: String) -> NSFetchRequest<Self> {
+        fetchRequest(keyPath: idKey, equalTo: id)
+    }
+    
+    static func fetchRequest(keyPath: String, equalTo value: String) -> NSFetchRequest<Self> {
+        let request = NSFetchRequest<Self>(entityName: entityName)
+        request.predicate = NSPredicate(format: "%K == %@", keyPath, value)
+        return request
+    }
+}
+
+extension NSManagedObject: NSFetchRequestGettable {}
+
 extension NSManagedObject {
-    static func load<T>(by request: NSFetchRequest<T>, context: NSManagedObjectContext) -> [T] {
+    static func load<T: NSManagedObject>(by id: String, context: NSManagedObjectContext) -> [T] {
+        load(keyPath: idKey, equalTo: id, context: context)
+    }
+    
+    static func load<T: NSManagedObject>(keyPath: String, equalTo value: String, context: NSManagedObjectContext) -> [T] {
+        let request = NSFetchRequest<T>(entityName: entityName)
+        request.predicate = NSPredicate(format: "%K == %@", keyPath, value)
+        return load(by: request, context: context)
+    }
+    
+    static func load<T: NSManagedObject>(by request: NSFetchRequest<T>, context: NSManagedObjectContext) -> [T] {
+        request.entity = NSEntityDescription.entity(forEntityName: T.entityName, in: context)!
         do {
-            return try context.fetch(request)
+            return try context.fetch(
+                request,
+                using: FetchCache.shared
+            )
```

Comment:
Why is this not in the same line?

Reply:
i've separated the lines to easily comment line 59 to test non-fetch performance. can be on the same line